src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -101,6 +108,7 @@ public Connection getReadConnection() throws SQLException {
      */
     public Connection getWriteConnection() throws SQLException {
         final State stateBefore = getState();
+        replicaConsistent = true;
```

Comment:
This looks weird to me.
Like you getting connection - it's got the operation and should be independent, as it won't make it pure function.
Maybe this is expected, but looking at this for the first time looks like a design smell and I've thought I'll share that thinking with you.

For me, this might be a suggestion to expand the logic behind replicaConsistent and doesn't have it as a single boolean value.